backends/dpdk/dpdkArch.cpp

```diff
@@ -80,25 +80,31 @@ const IR::Type_Control* ConvertToDpdkArch::rewriteControlType(const IR::Type_Con
 
 const IR::Type_Control* ConvertToDpdkArch::rewriteDeparserType(const IR::Type_Control* c,
                                                                cstring name) {
+    // Dpdk requires all local variables to be collected in a structure called
+    // metadata, and we perform read/write on this struct fields, so here we unify
+    // direction of this metadata decl to always be inout
     auto applyParams = new IR::ParameterList();
     if (name == "IngressDeparser") {
         applyParams->push_back(c->applyParams->parameters.at(0));
         auto header = c->applyParams->parameters.at(4);
         applyParams->push_back(new IR::Parameter(IR::ID("h"), header->direction, header->type));
         auto meta = c->applyParams->parameters.at(5);
-        applyParams->push_back(new IR::Parameter(IR::ID("m"), meta->direction, meta->type));
+        applyParams->push_back(
+            new IR::Parameter(IR::ID("m"), /*meta->direction*/ IR::Direction::InOut, meta->type));
     } else if (name == "EgressDeparser") {
         applyParams->push_back(c->applyParams->parameters.at(0));
         auto header = c->applyParams->parameters.at(3);
         applyParams->push_back(new IR::Parameter(IR::ID("h"), header->direction, header->type));
         auto meta = c->applyParams->parameters.at(4);
-        applyParams->push_back(new IR::Parameter(IR::ID("m"), meta->direction, meta->type));
+        applyParams->push_back(
+            new IR::Parameter(IR::ID("m"), /*meta->direction*/ IR::Direction::InOut, meta->type));
     } else if (name == "MainDeparserT") {
         applyParams->push_back(c->applyParams->parameters.at(0));
         auto header = c->applyParams->parameters.at(1);
         applyParams->push_back(new IR::Parameter(IR::ID("h"), header->direction, header->type));
         auto meta = c->applyParams->parameters.at(2);
-        applyParams->push_back(new IR::Parameter(IR::ID("m"), meta->direction, meta->type));
+        applyParams->push_back(
+            new IR::Parameter(IR::ID("m"), /*meta->direction*/ IR::Direction::InOut, meta->type));
```

Comment:
I think it is ok to remove the commented meta->direction

Reply:
ok